profitbricks/src/main/java/org/jclouds/profitbricks/compute/ProfitBricksComputeServiceAdapter.java

```diff
@@ -0,0 +1,267 @@
+	      api, ProvisioningStatusAware.DATACENTER, ProvisioningState.AVAILABLE), 10l * 60l, 2l, TimeUnit.SECONDS);
+   }
+
+   @Override
+   public NodeAndInitialCredentials<Server> createNodeWithGroupEncodedIntoName(String group, String name, Template template) {
+      String dataCenterId = template.getLocation().getId();
+      Hardware hardware = template.getHardware();
+
+      waitDcUntilAvailable.apply(dataCenterId);
+
+      List<? extends Volume> volumes = hardware.getVolumes();
+      List<String> storageIds = Lists.newArrayListWithExpectedSize(volumes.size());
+      int i = 1;
+      for (Volume volume : volumes) {
+	 logger.trace("<< provisioning storage '%s'", volume);
+	 String id = api.storageApi().createStorage(
+		 Storage.Request.creatingBuilder()
+		 .dataCenterId(dataCenterId)
+		 // put image to first storage
+		 .mountImageId(i == 1 ? template.getImage().getId() : "")
+		 //.imagePassword("<password_here>") // TODO consult jclouds committers
+		 .name(format("%s-disk-%d", name, i++))
+		 .size(volume.getSize())
+		 .build());
+	 waitDcUntilAvailable.apply(dataCenterId);
+	 storageIds.add(id);
+	 logger.trace(">> provisioning complete for storage. returned id='%s'", id);
+      }
+
+      Double cores = 0d;
+      for (Processor processor : hardware.getProcessors())
+	 cores += processor.getCores();
+
+      String storageBootDeviceId = Iterables.getFirst(storageIds, "");
+      Server.Request.CreatePayload serverRequest = Server.Request.creatingBuilder()
+	      .dataCenterId(dataCenterId)
+	      .name(name)
+	      .bootFromStorageId(storageBootDeviceId)
+	      .cores(cores.intValue())
+	      .ram(hardware.getRam())
+	      .availabilityZone(AvailabilityZone.AUTO)
+	      .hasInternetAccess(true)
+	      .build();
+
+      logger.trace("<< provisioning server '%s'", serverRequest);
+      String serverId = api.serverApi().createServer(serverRequest);
+      waitDcUntilAvailable.apply(dataCenterId);
+      logger.trace(">> provisioning complete for server. returned id='%s'", serverId);
+
+      final int storageCount = storageIds.size();
+      for (int j = 1; j < storageCount; j++) { // skip first; already connected
+	 String storageId = storageIds.get(j);
+	 logger.trace("<< connecting storage '%s' to server '%s'", storageId, serverId);
+	 api.storageApi().connectStorageToServer(
+		 Storage.Request.connectingBuilder()
+		 .storageId(storageId)
+		 .serverId(serverId)
+		 .build());
+
+	 waitDcUntilAvailable.apply(dataCenterId);
+	 logger.trace("<< storage connected.");
+      }
+
+      Server server = getNode(serverId);
+
+      // TODO Consult jclouds committers. Server credentials in profitbricks are sent 
+      // via email used in credentials.identity. We can set a password during 
+      // storage provisioning, maybe generate random characters?
+      LoginCredentials serverCredentials = LoginCredentials.builder()
+	      .user("root")
+	      .password("<password_here>")
+	      .build();
```

Comment:
jclouds needs the node credentials to be able to connect to the node when using the "runScript" methods. If credentials are configured per-node (instead of having default credentials per image), we should return them here.

If the password can be set during the node provisioning, that's perfect. If the user has set a password using the `templateOptions.overrideLoginPassword`, let's use that one, otherwise auto-generate a random one. The same applies to the user.